thunder/core/transforms.py

```diff
@@ -853,6 +853,21 @@ def _gather_prim_grad(a: TensorProxy, index: TensorProxy, dim: int) -> TensorPro
 register_grad(pids.GATHER, _gather_prim_grad)
 
 
+@torchctx
+def _scatter_add_prim_grad(a: TensorProxy, /, index: TensorProxy, value: TensorProxy, dim: int) -> TensorProxy:
+    fwd = prims.scatter_add(a, index, value, dim)
+
+    g = get_grad(fwd)
+    # NOTE The value gradient is only valid when src.shape == index.shape.
```

Comment:
(which was checked in forward)

Reply:
No, forward does not have this lim. This is the "hidden" requirement to make sure the grad is correct. It is an issue with a long history :)